src/main/scala/org/sunbird/cloud/storage/service/GcloudStorageService.scala

```diff
@@ -85,4 +91,49 @@ class GcloudStorageService(config: StorageConfig) extends BaseStorageService  {
     } else
       throw new StorageServiceException("uri not available for the given prefix: "+ _prefix)
   }
+
+  override def getSignedURLV4(container: String, objectKey: String, permission: Option[String] = Option("r"), ttl: Option[Int],
```

Comment:
@shishirsuman092 We do not need to have a getSignedURLV4 here and in the BaseStorageService.scala. The getSignedURLV2 was created to switch between getSignedURL (for other clouds) and getPutSignedURL for GCP. 

cc: @AmiableAnil @sharathkashyap @maheshkumargangula  From KP, we should use the getSignedURLV2 method so that we dont have cloud specific behavior in KP

Reply:
If any new parameters are required for GCP, these should be passed as optional items in the V2 method